llvm/lib/Target/AIE/AIEPostPipeliner.cpp

```
@@ -662,7 +662,12 @@ bool PostPipeliner::scheduleFirstIteration(PostPipelinerStrategy &Strategy) {
     MachineInstr *MI = SU.getInstr();
     const int Earliest = Strategy.earliest(SU);
     const int Latest = Strategy.latest(SU);
-    // Find the first cycle that fits. We try every position modulo II
+    if (Earliest > Latest) {
```

This was insidious. We would actually revert the insertion direction, since fit() computes that from the range. This would allow a latency violation, only detected when checking the other iterations. So the error was masked by an extra check. We short-circuit all of that by detecting an empty interval early.